### DISCUSSION TITLE
[FEEDBACK]: Leaking pipeline is missing

### DISCUSSION THREAD

**Discussion by robvanderveer** [OWASP/www-project-machine-learning-security-top-10#108] [Machine Learning Security Topics]
### Type

General Feedback

### What would you like to report?

The risk of leaking training data or other confidentiality issues of the AI pipeline (code, model parameters) are missing. 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by aryanxk02** [comment]
I can come up with some content associated with this topic, but where in the project do we integrate this exactly?
cc: @shsingh @robvanderveer

**Reply by shsingh** [reply]
hi @aryanxk02 if you want to take a stab at creating the content first let's just work on that for now. we can work out where it goes as part of this discussion. /cc @sagarbhure @mik0w @robvanderveer

**Comment by aryanxk02** [comment]
Hi @shsingh, this is a brief outline for the content associated with the above topic. Feel free the make changes wherever you think necessary :)

## Introduction
- Significance of data and model confidentiality in AI.

## Understanding AI pipeline
- Various stages of an AI pipeline, from data collection to model deployment.
- Importance of sensitivity of the data and model parameters at every stage.

## Critical Issue 1: Data Leakage
- What is Data Leakage?
- Possibility of data leakage occurrence and consequences.
- Prevention: Best practices.
- Real world examples of data leakage incidents.

## Critical Issue 2: Model Parameters
- Confidentiality of model parameters.
- Best practices to follow to protect the parameters.
- Importance of code review for model related code.


## Conclusion and References
- Information in a nutshell and cite references.

cc: @robvanderveer @sagarbhure

**Reply by robvanderveer** [reply]
This is a good start. Have a look at the AI guide for further inspiration.
You need to add 'code, documentation and configuration' as well, as that is also intellectual property right.
If you add this as a new entry in the top 10, you can focus 'model stealing' on behavioral stealing and add a remark there that another way to steal a model is to steel the parameters and then refer to this pipeline entry.

**Reply by aryanxk02** [reply]
Can you please share the link to the AI guide so I can refer to the same.

**Reply by robvanderveer** [reply]
https://owasp.org/www-project-ai-security-and-privacy-guide/
also listen to: https://mlsecops.com/podcast/a-holistic-approach-to-understanding-the-ai-lifecycle-and-securing-ml-systems-protecting-ai-through-people-processes-technology
watch: https://www.brighttalk.com/webcast/19697/586526
and look at especially the annexes in : https://www.enisa.europa.eu/publications/securing-machine-learning-algorithms
